### PR TITLE
Initial release of Ghost Palace Memorial

### PR DESCRIPTION
**Pull request by Atemo**
* **Addressed Issue(s)**: none

* **Server Mode**: renewal

* **Description of Pull Request**: Ghost Palace Memorial

Thanks @Lemongrass3110 for his help!
Thanks to Divine Pride for the monsters datas !

### CODE REVIEW

File: npc/re/instances/GhostPalace.txt [hunk 1]
```diff
@@ -0,0 +1,975 @@
...
+	mes "Okay, that is good. A traveller. Did you see?";
+	npctalk "Sakray: Okay, that is good. A traveller. Did you see?";
+	next;
+	mes "[" + strcharinfo(0) + "]";
+	mes "Sakray, What do you want sympathy? Or is it a trap?";
+	unittalk getcharid(3), "" + strcharinfo(0) + " : Sakray, What do you want sympathy? Or is it a trap?";
+	next;
+	mes "[Sakray]";
+	mes "None...I did not need to make an effort to catch a novice.";
+	npctalk "Sakray: None...I did not need to make an effort to catch a novice.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `No... I do not need to make any effort to catch a novice.`

File: npc/re/instances/GhostPalace.txt [hunk 2]
```diff
@@ -0,0 +1,975 @@
...
+	mes "If so... then why...?";
+	unittalk getcharid(3), "" + strcharinfo(0) + " : If so... then why...?";
+	next;
+	cutin "cry-b.bmp",2;
+	mes "[Sakray]";
+	mes "I wanted to get rid of a piece of humanity inside of me through you.";
+	npctalk "Sakray: I wanted to get rid of a piece of humanity inside of me through you.";
+	next;
+	mes "[Sakray]";
+	mes "I am quite satisfied with the result.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: Missing `npctalk` for here and below for Sakray?

File: npc/re/instances/GhostPalace.txt [hunk 3]
```diff
@@ -0,0 +1,975 @@
...
+	next;
+	mes "[Sakray]";
+	mes "Can you set yourself free from this curse?";
+	next;
+	mes "[" + strcharinfo(0) + "]";
+	mes "...";
+	unittalk getcharid(3), "" + strcharinfo(0) + " : ...";
+	next;
+	mes "[Sakray]";
+	mes "Heh heh... I will keep watching you until the day you grow up enough.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Heh heh... I will keep watching you until the day comes when you are old enough.`

File: npc/re/instances/GhostPalace.txt [hunk 4]
```diff
@@ -0,0 +1,975 @@
...
+	initnpctimer;
+	end;
+OnTimer3000:
+	npctalk "Voice of Princess Tiara : What can I do for you?";
+	end;
+OnTimer8000:
+	npctalk "Voice of Sakray : I will always stand by you. Please rest beside me.", instance_npcname("#sv");
+	end;
+OnTimer13000:
+	npctalk "Thanatos Magic Trace: Sakray, Do not forget the covenant of blood between us.", instance_npcname("#tv");
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Sakray, do not forget the covenant of blood between us.`

File: npc/re/instances/GhostPalace.txt [hunk 5]
```diff
@@ -0,0 +1,975 @@
...
+
+1@spa,195,41,1	duplicate(Lurid Royal Guard#dk)	Thanatos Magic Trace#dk3	CLEAR_NPC
+1@spa,198,43,1	duplicate(Lurid Royal Guard#dk)	#sv	CLEAR_NPC
+1@spa,196,41,1	duplicate(Lurid Royal Guard#dk)	#tv	CLEAR_NPC
+
+1@spa,204,29,1	script	King#dk11	4_M_RUSKING,{
+	disable_items;
+	mes "[King]";
+	mes "Hey there, are you alive?";
+	mes "I did not expect to meet someone alive";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `I did not expect to meet someone who is still alive.`

File: npc/re/instances/GhostPalace.txt [hunk 6]
```diff
@@ -0,0 +1,975 @@
...
+	next;
+	mes "[King]";
+	mes "Do you possibly have ^006400Gray Shard^000000?";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Do you possibly have the ^006400Gray Shard^000000?`

File: npc/re/instances/GhostPalace.txt [hunk 7]
```diff
@@ -0,0 +1,975 @@
...
+	mes "Can you give it to me?";
+	mes "The old memory of faituful royal guard,";
+	mes "the memory of everyone he loved";
+	mes "becomes broken pieces and are contained inside of it.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `The memories of the faithful royal guard`
`and the memories of everyone he loved`
`are contained inside of these shards.`

File: npc/re/instances/GhostPalace.txt [hunk 8]
```diff
@@ -0,0 +1,975 @@
...
+	next;
+	mes "[King]";
+	mes "Will you give ^006400Gray Shard^000000 to me?";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Will you give the ^006400Gray Shard^000000 to me?`

File: npc/re/instances/GhostPalace.txt [hunk 9]
```diff
@@ -0,0 +1,975 @@
...
+	mes "The guilt is eating my soul.";
+	mes "My soul is stuck in ghost palace.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `My soul is stuck within the Ghost Palace.`

File: npc/re/instances/GhostPalace.txt [hunk 10]
```diff
@@ -0,0 +1,975 @@
...
+	for ( .@i = 0; .@i < 20; .@i++ )
+		.@menu$ += .@item_name$[.@i] + ":";
+	.@menu$ += "Cancel";
+
+	while(true) {
+		.@s = select(.@menu$) - 1;
+		if (.@s == 20) {
+			mes "[King]";
+			mes "Someday, Sakray will";
+			mes "get out of curse for sure.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `get out of this curse for sure.`

File: npc/re/instances/GhostPalace.txt [hunk 11]
```diff
@@ -0,0 +1,975 @@
...
+			mes "I believe";
+			mes "^006400Gray Shard^000000 will be the only method";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `^006400Gray Shard^000000s are the only method`

File: npc/re/instances/GhostPalace.txt [hunk 12]
```diff
@@ -0,0 +1,975 @@
...
+			mes "[King]";
+			mes "Someday, Sakray will";
+			mes "get out of curse for sure.";
+			mes "I believe";
+			next;
+			continue;
+		}
+		if (countitem(6672) < .@cost) {
+			mes "[King]";
+			mes "Not enough ^006400Gray Shard^000000..";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `You do not have enough ^006400Gray Shard^000000`

File: npc/re/instances/GhostPalace.txt [hunk 13]
```diff
@@ -0,0 +1,975 @@
...
+			mes "to make a ^FF0000" + .@item_name$[.@s] + "^000000";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `to make a ^FF0000" + .@item_name$[.@s] + "^000000.`

File: npc/re/instances/GhostPalace.txt [hunk 14]
```diff
@@ -0,0 +1,975 @@
...
+		}
+		mes "[King]";
+		mes "I will make a ^FF0000" + .@item_name$[.@s] + "^000000";
+		mes "with " + .@cost + " ^006400Gray Shard^000000.";
+		next;
+		mes "You see a light through the folded hands of the king and it gradually takes on a shape.";
+		next;
+		mes "[King]";
+		mes "All right, it is done.";
+		mes "Take this ^FF0000" + .@item_name$[.@s] + "^000000.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Take this ^FF0000 + .@item_name$[.@s] + "^000000`

File: npc/re/instances/GhostPalace.txt [hunk 15]
```diff
@@ -0,0 +1,975 @@
...
+		delitem 6672,.@cost;// Gray Shard
+		getitem .@items_list[.@s],1;
+		close;
+	}
+}
+
+1@spa,210,28,0	script	Way out of Palace	WARPNPC,2,2,{
+	end;
+OnTouch_:
+	mes "It is time to leave Palace of the ghost.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `It is time to leave the Ghost Palace.`

File: npc/re/instances/GhostPalace.txt [hunk 16]
```diff
@@ -0,0 +1,975 @@
...
+	mes "Everything is completed,";
+	mes "Everything will be vanished..";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `Everything is now complete`
`and will vanish..`

File: npc/re/instances/GhostPalace.txt [hunk 17]
```diff
@@ -0,0 +1,975 @@
...
+		mes "Have you found the princess yet?! Is she safe?";
+		close;
+	case 2:
+		mes "[Unpleasent Royal Guard]";
+		mes "The monsters around the secret passage have disappeared. Now is our chance to save the princess!";
+		erasequest 1261;// Cursed Swordman
+		close;
+	}
+	mes "[Unpleasent Royal Guard]";
+	mes "I need some help; even from the likes of you.";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `I need help, even from the likes of you.`

File: npc/re/instances/GhostPalace.txt [hunk 18]
```diff
@@ -0,0 +1,975 @@
...
+	if (BaseLevel < 120) {
+		mes "[Unpleasent Royal Guard]";
+		mes "Althought I need help with a task, you are not strong enough to assist me, noob~!";
+		next;
+		mes "[Unpleasent Royal Guard]";
+		mes "Come back after when you have done some training and are at least level 120!";
+		close;
+	}
+	mes "[Unpleasent Royal Guard]";
+	mes "I can see the secret passage into the Palace... Are you ready to save the princess?!";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: `I can see the secret passage into the palace. Are you ready to save the princess?!`

File: npc/re/instances/GhostPalace.txt [hunk 19]
```diff
@@ -0,0 +1,975 @@
...
+	mes "I will always stand by you and protect you forever.";
+	npctalk "Lurid Royal Guard: I will always stand by you and protect you forever.";
+	cutin "npc-tiara.bmp",255;
+	next;
+	mes "[Princess Tiara]";
+	mes "Monsters in the castle? Protect me!";
+	npctalk "Princess Tiara: Monsters in the castle? Protect me!", .@tiara$;
+	next;
+	mes "[Lurid Royal Guard]";
+	mes "Step back, Let's go!";
```
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: Missing `npctalk`?

File: npc/re/instances/GhostPalace.txt (hunk 14 above)
Comment (Atemo) [inline npc/re/instances/GhostPalace.txt]: There is a mistake here?

File: npc/re/instances/GhostPalace.txt (hunk 2 above)
Comment (Atemo) [inline npc/re/instances/GhostPalace.txt]: I forgot them yes :o

File: npc/re/instances/GhostPalace.txt [hunk 20]
```diff
@@ -0,0 +1,975 @@
...
+	cutin "npc-tiara.bmp",3;
+	mes "[Princess Tiara]";
+	mes "Thank you... now I can rest beside you...";
+	npctalk "Princess Tiara: Thank you... now I can rest beside you...";
+	next;
+	mes "[Lurid Royal Guard]";
+	mes "No..No!!!!!";
+	npctalk "Lurid Royal Guard: No..No!!!!!", .@guard$;
+	next;
+	mes "[The voice of Princess Tiara]";
```
Comment (Atemo) [inline npc/re/instances/GhostPalace.txt]: Well yes the header change here and the npc is disabled at the end.

File: npc/re/instances/GhostPalace.txt (hunk 14 above)
Comment (aleos89) [inline npc/re/instances/GhostPalace.txt]: To remove the period at the end to continue the sentence on to the next `mes`.

File: npc/re/instances/GhostPalace.txt [hunk 21]
```diff
@@ -0,0 +1,982 @@
...
+		"Gray Shield", "Gray Armor", "Gray Robe", "Gray Cloak", "Gray Shoes", "Gray Helmet";
+	setarray .@items_list[0],13441,21009,1438,1496,1669,2023,18119,13093,28000,1836,16028,28100,1933,1988,2187,15090,15091,20721,22033,18820;
+
+	for ( .@i = 0; .@i < 20; .@i++ )
+		.@menu$ += .@item_name$[.@i] + ":";
+	.@menu$ += "Cancel";
+
+	while(true) {
+		.@s = select(.@menu$) - 1;
+		if (.@s == 20) {
```
Comment (Jeybla) [inline npc/re/instances/GhostPalace.txt]: `.@s == 20` -> `.@s < 0 || .@s >= 20` if anything goes wrong. Maybe It'd be better to replace the hardcoded `20` with something like `.@item_cnt`

Comment (Atemo) [inline npc/re/instances/GhostPalace.txt]: hmm it's fine for me. A static list build the menu and the menu never change

**Pull request merged**